Hazelcast.Test/Hazelcast.Client.Test/PredicatesTest.cs

```diff
@@ -0,0 +1,443 @@
+        }
+
+        [Test]
+        public virtual void TestLessThanOrEqual()
+        {
+            FillMap();
+            var predicate = CreateLessThanOrEqual();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-0", "key-1"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestLike()
+        {
+            map.Put("key-1", "a_value");
+            map.Put("key-2", "b_value");
+            map.Put("key-3", "aa_value");
+            map.Put("key-4", "AA_value");
+            var predicate = CreateLike();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-1", "key-3"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestNot()
+        {
+            FillMap();
+            var predicate = CreateNot();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-0"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestNotEqual()
+        {
+            FillMap();
+            var predicate = CreateNotEqual();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-1", "key-2", "key-3", "key-4", "key-5", "key-6", "key-7", "key-8", "key-9"},
+                Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestOr()
+        {
+            FillMap();
+            var predicate = CreateOr();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-1", "key-2"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestRegex()
+        {
+            map.Put("key-1", "car");
+            map.Put("key-2", "cry");
+            map.Put("key-3", "giraffe");
+            var predicate = CreateRegex();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-1", "key-2"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestSql()
+        {
+            FillMap();
+            var predicate = CreateSql();
+            var keySet = map.KeySet(predicate);
+            Assert.That(new[] {"key-1"}, Is.EquivalentTo(keySet.ToArray()));
+        }
+
+        [Test]
+        public virtual void TestTrue()
+        {
+            FillMap();
+            var predicate = CreateTrue();
+            var keySet = map.KeySet(predicate);
+            Assert.That(map.KeySet(), Is.EquivalentTo(keySet.ToArray()));
+        }        
+        
+        [Test]
+        public virtual void TestAnd2()
+        {
+            var predicate = CreateAndPredicate();
+            map.Put("key", predicate);
+            map.Put("key", predicate);
+            Assert.AreEqual(predicate, map.Get("key"));
+        }
+
+        [Test]
+        public virtual void TestBetween2()
```

Comment:
could use more descriptive name for test here..  like TestBetween_Deserialization or something similar